Arjun/Warehouse Winforms Project/Warehouse/ManagementApp/frmPickProduct.cs

```diff
@@ -0,0 +1,295 @@
+
+                double tempDistance = FindDistance(tempArray[4], originCoordinates);
+                tempArray[8] = (tempDistance).ToString();
+                distanceToIndex.Add(tempDistance, i);
+
+                Random r = new Random();      //generating a random number for Quantity
+                randomNum = r.Next(1, 10);
+
+                tempArray[9] = randomNum.ToString();
+
+                productList.Add(tempArray);
+            }
+
+            foreach (KeyValuePair<double, double> blabla in distanceToIndex)
+            {
+                sortedList.Add(blabla.Value);
+            }
+
+            leastDistanceIndex = Convert.ToInt32(sortedList[0]);
+
+            SetProductPropertiesAndDisplay(productList);
+
+            totalCost = product.Cost;
+
+            dba.UpdateProductData(product);
+
+            productListUpdated.Add(productList[leastDistanceIndex]);
+
+            for (int i = 0; i < productList.Count; i++)
+            {
+                if (i != leastDistanceIndex)
+                {
+                    productListUpdated.Add(productList[i]);
+                }
+            }
+
+        }
+
+        private void SetProductPropertiesAndDisplay(List<string[]> productList)
+        {
+            lblItemName.Text = productList[leastDistanceIndex][0];
+            lblLocation.Text = productList[leastDistanceIndex][3];
+            lblQuantity.Text = productList[leastDistanceIndex][9];
+
+
+            product.Name = productList[leastDistanceIndex][0];
+            product.Dpci = productList[leastDistanceIndex][1];
+            product.UnitPrice = Convert.ToDouble(productList[leastDistanceIndex][2].ToString());
+            product.Quantity = Convert.ToDouble(productList[leastDistanceIndex][9]);
+            product.Cost = product.UnitPrice * product.Quantity;
+            
+
+
+            product.FullDetails = string.Empty;
+            product.FullDetails = $"Item: {product.Name}\t" +
+                                      $"Unit Price: {product.UnitPrice}\t" +
+                                      $"Units: {product.Quantity} \t" +
+                                      $"Cost: {product.Cost}";
+
+            product.Summary.Add(product.FullDetails);
+            productCount++;
+
+            product.Availability = Convert.ToDouble(productList [leastDistanceIndex][5].ToString());
+
+            if (product.Availability >= product.Quantity)
+            {
+                product.Availability = product.Availability - product.Quantity;
+            }
+
+            else
+            {
+                product.Quantity = product.Availability;
+                product.Availability = 0;
+                //need to figure out what to do when products are unavailable
+                //Probably there must be some message.
+            }
+        }
+
+
+        private static double FindDistance(string v, string w)
+        {
+            string x1 = v.Substring(0, v.IndexOf(","));

```

Comment:
This is inefficient and also it ties the representation of a type to logic. What if the format for the string was changed? Then you would have to change the logic too 
Instead declare a Vector2 or use the built in Vector2 that can be parsed from a string